src/runtime/offers-flow.js

```diff
@@ -30,41 +46,53 @@ const offersUrl = '$frontend$/subscribewithgoogleclientui/offersiframe';
 export class OffersFlow {
 
   /**
-   * @param {!Window} win
-   * @param {!../model/page-config.PageConfig} pageConfig
-   * @param {!web-activities/activity-ports.ActivityPorts} activityPorts
+   * @param {!../model/deps.DepsDef} deps
    */
-  constructor(win, pageConfig, activityPorts) {
+  constructor(deps) {
+
+    /** @private @const {!../model/deps.DepsDef} */
+    this.deps_ = deps;
+
     /** @private @const {!Window} */
-    this.win_ = win;
+    this.win_ = deps.win();
 
     /** @private @const {!HTMLDocument} */
-    this.document_ = win.document;
+    this.document_ = this.win_.document;
 
     /** @private @const {!web-activities/activity-ports.ActivityPorts} */
-    this.activityPorts_ = activityPorts;
+    this.activityPorts_ = deps.activities();
 
-    /** @private @const {!Dialog} */
-    this.dialog_ = new Dialog(this.win_);
+    /** @private @const {!../components/dialog-manager.DialogManager} */
+    this.dialogManager_ = deps.dialogManager();
 
     /** @private @const {!ActivityIframeView} */
     this.activityIframeView_ = new ActivityIframeView(
         this.win_,
         this.activityPorts_,
-        offersUrl,
+        OFFERS_URL,
         {
-          'publicationId': pageConfig.getPublicationId(),
-          'label': pageConfig.getLabel(),
+          'publicationId': deps.pageConfig().getPublicationId(),
+          'label': deps.pageConfig().getLabel(),
         });
+
+    PayCompleteFlow.configurePending(this.deps_);
```

Comment:
@dparikh quick note: `configurePending` can only be called from main runtime since only one callback per page is needed.